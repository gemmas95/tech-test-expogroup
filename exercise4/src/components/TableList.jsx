import React from "react";
import Contributor from "./Contributor";

function TableList({ contributorsList }) {
  return (
    <>
      {!contributorsList && (
        <p className="text-center m-5">
          Please enter a repository name owned by SkylabCoders to find it's
          contributors <span>(suggestions: skylab-bootcamp-202007)</span>
        </p>
      )}
      {contributorsList?.length === 0 && (
        <img
          alt="loading"
          src="https://trello-attachments.s3.amazonaws.com/5f7c8ab9b80a927f1f047d20/300x300/a35a54d93989cc51ca3226d2220477b3/Gif_loaging.gif"
        ></img>
      )}
      {contributorsList?.length > 0 && contributorsList[0].login && (
        <ul className="card-deck flex-wrap cards__container">
          {contributorsList?.map((user) => (
            <Contributor user={user} key={user.id} />
          ))}
        </ul>
      )}
      {contributorsList?.message && (
        <p className="text-center m-5 alert alert-danger">
          Error: {contributorsList.message}
        </p>
      )}
    </>
  );
}

export default TableList;
