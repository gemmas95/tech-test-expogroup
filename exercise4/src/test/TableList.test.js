import renderer from "react-test-renderer";
import React from "react";
import TableList from "../components/TableList";

describe("TableList", () => {
  let fakecontributorsList = null;
  it("should match", () => {
    const tableListTree = renderer.create(
      <TableList contributorsList={fakecontributorsList} />
    );
    expect(tableListTree).toMatchSnapshot();
  });
  it("should match", () => {
    fakecontributorsList = [
      { login: "pepe", id: 24 },
      { login: "lola", id: 2 },
    ];
    const tableListTree = renderer.create(
      <TableList contributorsList={fakecontributorsList} />
    );
    expect(tableListTree).toMatchSnapshot();
  });
});
