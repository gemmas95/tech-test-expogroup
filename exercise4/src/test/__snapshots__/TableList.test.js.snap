// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableList should match 1`] = `
<p
  className="text-center m-5"
>
  Please enter a repository name owned by SkylabCoders to find it's contributors 
  <span>
    (suggestions: skylab-bootcamp-202007)
  </span>
</p>
`;

exports[`TableList should match 2`] = `
<ul
  className="card-deck flex-wrap cards__container"
>
  <li
    className="profile-card"
  >
    <img
      alt={24}
      data-testid="avatar_url"
      src="undefined"
    />
    <p
      className="text-uppercase text-monospace font-weight-bold"
      data-testid="user__login"
    >
      pepe
    </p>
    <p
      data-testid="contributions"
    >
      Number of contributions: 
    </p>
    <a
      className="btn btn-dark"
      data-testid="html_url"
    >
       
      Link to profile
    </a>
  </li>
  <li
    className="profile-card"
  >
    <img
      alt={2}
      data-testid="avatar_url"
      src="undefined"
    />
    <p
      className="text-uppercase text-monospace font-weight-bold"
      data-testid="user__login"
    >
      lola
    </p>
    <p
      data-testid="contributions"
    >
      Number of contributions: 
    </p>
    <a
      className="btn btn-dark"
      data-testid="html_url"
    >
       
      Link to profile
    </a>
  </li>
</ul>
`;
